test/unit/test.py

```diff
@@ -2262,7 +2318,55 @@ def test_notified_when_working_in_deleted_task(self):
             # load an edit page
             erica.open_link(url='/tree/{}/edit/other/'.format(erica_branch_name))
             # a warning is flashed about working in a deleted branch
-            self.assertIsNotNone(erica.soup.find(lambda tag: tag.name == 'li' and tag.text == u'This activity has been deleted or never existed! Please start a new activity to make changes.'))
+            self.assertIsNotNone(erica.soup.find(lambda tag: tag.name == 'li' and tag.text == view_functions.MESSAGE_ACTIVITY_DELETED))
+
+    # in TestProcess
+    def test_page_not_found_when_branch_deleted(self):
+        ''' When you're working in a deleted branch and don't have a local copy, you get a 404 error
+        '''
+        with HTTMock(self.auth_csv_example_allowed):
+            with HTTMock(self.mock_persona_verify_erica):
+                erica = ChimeTestClient(self.app.test_client(), self)
+                erica.sign_in('erica@example.com')
+
+            with HTTMock(self.mock_persona_verify_frances):
+                frances = ChimeTestClient(self.app.test_client(), self)
+                frances.sign_in('frances@example.com')
+
+            # Start a new task
+            task_description = u'Eating Carrion'
+            task_beneficiary = u'Vultures'
+            erica.start_task(description=task_description, beneficiary=task_beneficiary)
+            erica_branch_name = erica.get_branch_name()
+
+            # Enter the "other" folder
+            erica.follow_link(href='/tree/{}/edit/other/'.format(erica_branch_name))
+
+            #
+            # Switch users
+            #
+            # delete erica's task
+            frances.open_link(url='/')
+            frances.delete_task(branch_name=erica_branch_name)
+            self.assertEqual(PATTERN_FLASH_TASK_DELETED.format(description=task_description, beneficiary=task_beneficiary), frances.soup.find('li', class_='flash').text)
+
+            # delete all trace of the branch locally
+            repo = view_functions.get_repo(repo_path=self.app.config['REPO_PATH'], work_path=self.app.config['WORK_PATH'], email='erica@example.com')
+            repo.git.checkout('master')
+            repo.git.branch('-D', erica_branch_name)
+            repo.git.remote('prune', 'origin')
+
+            #
+            # Switch users
+            #
+            # load an edit page
+            erica.open_link(url='/tree/{}/edit/other/'.format(erica_branch_name), expected_status_code=404)
+            # a warning is flashed about working in a deleted branch
+            self.assertIsNotNone(erica.soup.find(lambda tag: tag.name == 'li' and tag.text == view_functions.MESSAGE_ACTIVITY_DELETED))
```

Comment:
This can also be `erica.soup.find(text=view_functions.MESSAGE_ACTIVITY_DELETED)` for brevity.